### PR TITLE
COMMERCE-2495 Added coupon code taglib

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

File: commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java [hunk 1]
```diff
@@ -180,6 +182,45 @@ public Response getOrder(
 		return getResponse(cart);
 	}
 
+	@Path("/apply-coupon-code/{orderId}/{couponCode}")
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response getOrder(
```
Comment (marco-leo) [inline commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java]: getOrder?

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/CouponCodeTag.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+public class CouponCodeTag extends IncludeTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		try {
+			CommerceContext commerceContext =
+				(CommerceContext)request.getAttribute(
+					CommerceWebKeys.COMMERCE_CONTEXT);
+
+			_commerceOrder = commerceContext.getCommerceOrder();
```
Comment (marco-leo) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/CouponCodeTag.java]: pass order Id by taglib parameter

File: commerce-frontend-taglib/src/main/resources/META-INF/resources/coupon_code/page.jsp [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+				<liferay-ui:icon
+					icon="times"
+					markupView="lexicon"
+					message="remove"
+				/>
+			</a>
+		</div>
+
+		<aui:script use="aui-base">
+			A.one('#<portlet:namespace />couponCodeIconRemove').on(
```
Comment (marco-leo) [inline commerce-frontend-taglib/src/main/resources/META-INF/resources/coupon_code/page.jsp]: avoid usage of Alloy

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java (hunk 1 above)
Comment (alessiorendina) [reply commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/CommerceCartResource.java]: Changed in applyCouponCode

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/CouponCodeTag.java (hunk 2 above)
Comment (alessiorendina) [reply commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/CouponCodeTag.java]: Did it, thanks

File: commerce-frontend-taglib/src/main/resources/META-INF/resources/coupon_code/page.jsp (hunk 3 above)
Comment (alessiorendina) [reply commerce-frontend-taglib/src/main/resources/META-INF/resources/coupon_code/page.jsp]: Did it aswell

**Comment by alessiorendina** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test

**Comment by marco-leo** [comment]
@gianmarcobrunialti this should be transformed in a InfoRenderComponent

**Comment by gianmarcobrunialti** [comment]
@marco-leo Alright.

**Pull request merged**